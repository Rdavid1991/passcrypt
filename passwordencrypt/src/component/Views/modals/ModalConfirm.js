import React from 'react';
import { Button } from '../../common';

export const ModalConfirm = ({handleCancel,existEmail}) => {
    return (
        <div className="modal fade" id="confirm">
            <div className="modal-dialog" style={{ top: '25%' }}>
                <div className="modal-content">
                    <div className="modal-body">
                        <h3 className="text-center">
                            { existEmail? "El email existe"  : "Se registro satisfactoriamente"}
                        </h3>
                        <div className="d-flex justify-content-center">
                            <Button
                                title="OK"
                                className="btn btn-success"
                                dataDismiss="modal"
                                click={existEmail? ()=>{}  : handleCancel}
                            />
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
};
