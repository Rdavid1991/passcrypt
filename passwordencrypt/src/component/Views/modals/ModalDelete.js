import React from 'react';

// eslint-disable-next-line react/prop-types
export const ModalDelete = ({ handleServiceDelete }) => {
    return (
        <div id="modal-confirm" className="modal fade">
            <div className="modal-dialog" style={{ marginTop: '15%' }}>
                <div className="modal-content">
                    <div className="modal-body text-center">
                        <h2>Â¿Desea borrar?</h2>
                        <hr />
                        <div className="d-flex justify-content-around">
                            <button
                                data-target="#modal-confirm"
                                data-dismiss="modal"
                                className="btn btn-secondary w-25"
                            >
                                No
                            </button>
                            <button
                                data-target="#modal-confirm"
                                data-dismiss="modal"
                                className="btn btn-primary w-25"
                                onClick={handleServiceDelete}
                            >
                                Si
                            </button>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
};
