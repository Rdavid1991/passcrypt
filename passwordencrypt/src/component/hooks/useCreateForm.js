import { useState } from "react";


export const useCreateForm = ( initialState = {}) => {

    const [values, setValues] = useState( initialState );

    const reset = () => {
        setValues( initialState );
    };

    const setEditValue = ( values )=>{
        setValues( values );
    };

    const handleInputChange = ({ target }) => {
        setValues({
            ...values,
            [target.name]: target.value
        });
    };


    return [values, handleInputChange, reset, setEditValue];
};