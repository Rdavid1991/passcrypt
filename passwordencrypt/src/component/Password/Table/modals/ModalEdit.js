/* eslint-disable react/prop-types */
import React, { useEffect } from 'react';
import { AES } from 'crypto-js';
import { useCreateForm } from '../../../../hooks/useCreateForm';
import { ModalForm } from './ModalForm';
import { initEdit } from '../../../../helpers/helpers';

export const ModalEdit = ({ handleServiceEdit, services, itemCode }) => {

    const [inputChange, handleInputChange, setEditValue] = useCreateForm( initEdit( services,itemCode ));

    const { id, user, service, password } = inputChange;

    useEffect(() => {
        setEditValue( initEdit( services,itemCode ));
    }, [itemCode]);

    const handleSaveService = ( ) => {
        handleServiceEdit({
            id,
            user   ,
            service ,
            password
        });
    };

    return (
        <ModalForm
            handleSaveService={handleSaveService}
            handleInputChange={handleInputChange}
            inputChange={inputChange}
            htmlId={"modal-edit"}
        />
    );
};
