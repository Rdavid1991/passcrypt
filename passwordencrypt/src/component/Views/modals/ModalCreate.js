import React from 'react';
import { useCreateForm } from '../../hooks/useCreateForm';
import { ModalForm } from './ModalForm';

// eslint-disable-next-line react/prop-types
export const ModalCreate = ({ handleServiceAdd }) => {

    const [inputChange, handleInputChange, reset] = useCreateForm({
        service : '',
        user    : '',
        password: ''
    });

    const { user, service, password } = inputChange;

    const handleSaveService = () => {

        handleServiceAdd({
            id: new Date().getTime(),
            user,
            service,
            password,
        });

        reset();
    };

    return (
        <ModalForm
            handleSaveService={handleSaveService}
            handleInputChange={handleInputChange}
            inputChange={inputChange}
            htmlId={"modal-form"}
        />
    );
};
