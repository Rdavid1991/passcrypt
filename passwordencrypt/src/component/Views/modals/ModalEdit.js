/* eslint-disable react/prop-types */
import React, { useEffect } from 'react';
import { useCreateForm } from '../../hooks/useCreateForm';
import { ModalForm } from './ModalForm';
import { initEdit } from '../../../helpers/helpers';

//import PropType from 'prop-types';

export const ModalEdit = ({ handleServiceEdit, services, itemCode }) => {

    const [inputChange, handleInputChange, setEditValue] = useCreateForm( initEdit( services,itemCode ));

    const { id, user, service, password } = inputChange;

    useEffect(() => {
        setEditValue( initEdit( services,itemCode ));
    }, [itemCode]);



    const handleSaveService = ( e ) => {
        e.preventDefault();

        handleServiceEdit({
            id,
            user   ,
            service ,
            password
        });

        window.$( '#modal-edit' ).modal( 'hide' );
    };

    return (
        <ModalForm
            handleSaveService={handleSaveService}
            handleInputChange={handleInputChange}
            inputChange={inputChange}
            htmlId={"modal-edit"}
        />
    );
};

/* ModalEdit.prototype = {
    itemCode: PropType.string
} */
