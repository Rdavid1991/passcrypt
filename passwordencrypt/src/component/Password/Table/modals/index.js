import { ModalCreate } from './ModalCreate';
import { ModalDelete } from './ModalDelete';
import { ModalEdit } from './ModalEdit';

export {
    ModalCreate,
    ModalDelete,
    ModalEdit
};