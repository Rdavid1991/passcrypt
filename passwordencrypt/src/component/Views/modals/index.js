import { ModalCreate } from './ModalCreate';
import { ModalDelete } from './ModalDelete';
import { ModalEdit } from './ModalEdit';
import {GeneratePassword} from './ModalGeneratePwd';

export {
    ModalCreate,
    ModalDelete,
    ModalEdit,
    GeneratePassword
};