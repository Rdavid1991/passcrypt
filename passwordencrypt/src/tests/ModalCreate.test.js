import '@testing-library/jest-dom'
import React from 'react';
import { shallow } from 'enzyme';
import { ModalCreate } from "../component/Views/modals/ModalCreate"
import { useUserContext } from '../component/hooks/useUserContext';

describe('Pruebas en crear', () => {

    const wrapper = shallow(
        <useUserContext value={{
            user: {
                userName: "userTest",
                userPassword: "userPassword"
            }
        }}>
            <ModalCreate/>
        </useUserContext>
    )

    test('Debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })
})


