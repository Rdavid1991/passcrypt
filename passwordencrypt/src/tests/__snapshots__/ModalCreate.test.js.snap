// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en crear Debe mostrar el componente correctamente 1`] = `
<useUserContext
  value={
    Object {
      "user": Object {
        "userName": "userTest",
        "userPassword": "userPassword",
      },
    }
  }
>
  <ModalCreate />
</useUserContext>
`;
